### ISSUE TITLE
./LomionReader' returned a non-zero code: 139

### ISSUE THREAD

**Issue by larsjsol** [decheine/complete-compendium#242]
Hi,

I'm trying to run the monster harvester as described in the readme, but it crashes at step 1 with the message `ProcessingThe command '/bin/bash -c ./LomionReader' returned a non-zero code: 139`.

Is this a bug or am I screwing somtheing up?

This is on ubuntu 24.04 with Docker version `26.1.3, build 26.1.3-0ubuntu1~24.04.1+esm1`.

```
$ sudo docker build -t monster-harvester -f harvester/Dockerfile .
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            Install the buildx component to build images with BuildKit:
            https://docs.docker.com/go/buildx/

Sending build context to Docker daemon  4.054GB
Step 1/22 : FROM ubuntu:22.04
 ---> 97271d29cb79
Step 2/22 : LABEL Description="Monster Harvester Build Environment" Version="0.1"
 ---> Using cache
 ---> a1d05f6f007a
Step 3/22 : ENV HOME /root
 ---> Using cache
 ---> 51b69d7c3972
Step 4/22 : SHELL ["/bin/bash", "-c"]
 ---> Using cache
 ---> c10fef46f3fe
Step 5/22 : RUN apt-get clean
 ---> Using cache
 ---> 83c94e3051c1
Step 6/22 : RUN apt-get -y update
 ---> Using cache
 ---> 4bf670420168
Step 7/22 : RUN DEBIAN_FRONTEND=noninteractive apt-get -y --no-install-recommends install     build-essential     clang     cmake     git     gdb     wget
 ---> Using cache
 ---> 9814f44d8feb
Step 8/22 : RUN apt-get install -y ca-certificates
 ---> Using cache
 ---> 58cfeda91e8e
Step 9/22 : RUN git config --global http.sslCAinfo /etc/ssl/certs/ca-certificates.crt
 ---> Using cache
 ---> 900f9308303d
Step 10/22 : ADD harvester/cmm /usr/local/harvester/cmm
 ---> Using cache
 ---> 23294fc6776e
Step 11/22 : ADD harvester/harvester /usr/local/harvester/harvester
 ---> Using cache
 ---> d7e3395b2e33
Step 12/22 : ADD harvester/include /usr/local/harvester/include
 ---> Using cache
 ---> 39d131a3b8c2
Step 13/22 : ADD harvester/src /usr/local/harvester/src
 ---> Using cache
 ---> d8667f0fe69a
Step 14/22 : ADD /data /usr/local/harvester/data
 ---> Using cache
 ---> 504fc912a0b7
Step 15/22 : ADD harvester/CMakeLists.txt /usr/local/harvester/CMakeLists.txt
 ---> Using cache
 ---> 09c398f18e26
Step 16/22 : WORKDIR /usr/local/harvester
 ---> Using cache
 ---> 76c92484789f
Step 17/22 : RUN mkdir build
 ---> Using cache
 ---> e895b0b8bf5d
Step 18/22 : WORKDIR /usr/local/harvester/build
 ---> Using cache
 ---> 454fd3fbb174
Step 19/22 : RUN cmake ..
 ---> Using cache
 ---> 5a3c20a1795b
Step 20/22 : RUN cmake --build ./
 ---> Using cache
 ---> cef4f4ef54c8
Step 21/22 : WORKDIR /usr/local/harvester/build
 ---> Using cache
 ---> 1ac03afaeb04
Step 22/22 : RUN ./LomionReader
 ---> Running in 54c16c32d1fc
Reading monster files in /usr/local/harvester/build/bin/monster_files ...
Monster files read.
monster names:
Testing Character: ×
Harvesting all!
Harvesting All
making 7 sets of size 309 and one set of size 6 to fill 2478 items
set size 309
set size 309
launched thread. set size 309 
Processing monster: aarakath
set size 309
launched thread. set size 309 
Processing monster: catwinge
set size 309
launched thread. set size 309 
Processing monster: drnecndr
set size 309
launched thread. set size 309 
Processing monster: giancrag
set size 309
launched thread. set size 309 
Processing monster: jakar
set size 309
launched thread. set size 309 
Processing monster: mummybog
set size 315
launched thread. set size 309 
Processing monster: scro
joining thread 1
launched thread. set size 315 
Processing monster: thunchil
Processing monster: giancycl
Processing monster: drnefadr
Processing monster: cavefish
Processing monster: thunhead
Processing monster: jalathga
Processing monster: mummyice
Processing monster: scryxull
Processing monster: aarakocr
Processing monster: giandese
Processing monster: murdakus
Processing monster: drnefidr
Processing monster: jalebird
Processing monster: thylacin
Processing monster: cedaspaw
Processing monster: scylchar
Processing monster: aartuk
Processing monster: cenkdorv
Processing monster: gianetti
Processing monster: murderoi
Processing monster: tickhear
Processing monster: jammleec
Processing monster: seademon
Processing monster: drnegedr
Processing monster: cenkgnoa
Processing monster: tickler
Processing monster: gianfhoi
Processing monster: murska
Processing monster: drnemodr
Processing monster: jarbo
Processing monster: sealion
Processing monster: tiefling
Processing monster: gianfirb
Processing monster: cenkhapo
Processing monster: aasiagat
Processing monster: drnepadr
Processing monster: jelgiamy
Processing monster: seaspama
Processing monster: myconid
Processing monster: cenkzebr
Processing monster: gianfire
Processing monster: tigeebon
Processing monster: aasideva
Processing monster: drneppdr
Processing monster: jellgian
Processing monster: seaspami
Processing monster: centaur
Processing monster: gianfog
Processing monster: drnepsdr
Processing monster: timedime
Processing monster: jhakar
Processing monster: aasilite
Processing monster: seawyrm
Processing monster: myrlocha
Processing monster: centdese
Processing monster: gianform
Processing monster: drspawn
Processing monster: jishmush
Processing monster: aasimar
Processing monster: tinkerer
Processing monster: seedbrai
Processing monster: mythfolk
Processing monster: centiped
Processing monster: drspgren
Processing monster: gianfros
Processing monster: jollroge
Processing monster: naga
Processing monster: seemer
Processing monster: tirichik
Processing monster: aasimon
Processing monster: drsplemo
Processing monster: centnoma
Processing monster: gianhalf
Processing monster: jorri
Processing monster: aasiplan
Processing monster: nagabone
Processing monster: seemwalk
Processing monster: titan
Processing monster: dryad
Processing monster: gianheph
Processing monster: jozhal
Processing monster: aasisola
Processing monster: nagadark
Processing monster: duckbunn
Processing monster: segarran
Processing monster: juhrion
Processing monster: centspir
Processing monster: gianhill
Processing monster: aballin
Processing monster: tliz
Processing monster: nagpa
Processing monster: kada
Processing monster: duende
Processing monster: sehogian
Processing monster: gianisla
Processing monster: cerepara
Processing monster: abhir
Processing monster: tlokasaz
Processing monster: kaisharg
Processing monster: narek
Processing monster: aboleth
Processing monster: selkie
Processing monster: duhlarki
Processing monster: cetusles
Processing monster: gianjung
Processing monster: toadgian
Processing monster: nasnas
Processing monster: kala
Processing monster: aboleths
Processing monster: chac
Processing monster: selkstar
Processing monster: gianmoun
Processing monster: dunefrea
Processing monster: toadleec
Processing monster: nat
Processing monster: kalin
Processing monster: serplord
Processing monster: dunereap
Processing monster: changelc
Processing monster: gianogre
Processing monster: toadspin
Processing monster: nathri
Processing monster: abomdiir
Processing monster: kalmari
Processing monster: serpwing
Processing monster: gianreef
Processing monster: tohrkre1
Processing monster: changeli
Processing monster: dunerunn
Processing monster: abrian
Processing monster: nautgian
Processing monster: gianshad
Processing monster: shaaz
Processing monster: kalothag
Processing monster: dunestal
Processing monster: nawidneh
Processing monster: absalom
Processing monster: gianspse
Processing monster: tohrkre2
Processing monster: shad
Processing monster: chaobeas
Processing monster: kaluk
Processing monster: dunetrap
Processing monster: naychurr
Processing monster: abyssant
Processing monster: gianston
Processing monster: kamatlan
Processing monster: duruchil
Processing monster: shadasp
Processing monster: charunte
Processing monster: tohrkre3
Processing monster: necroslu
Processing monster: achaiera
Processing monster: gianstor
Processing monster: kamigoro
Processing monster: shaddrak
Processing monster: needlman
Processing monster: chathran
Processing monster: dusanu
Processing monster: actaeon
Processing monster: gianswam
Processing monster: tombtapp
Processing monster: nehthalg
Processing monster: dvati
Processing monster: chattur
Processing monster: kamummy
Processing monster: addazahr
Processing monster: shade
Processing monster: giantath
Processing monster: tombward
Processing monster: chevall
Processing monster: dwaraqua
Processing monster: neogi
Processing monster: kanidoll
Processing monster: adherer
Processing monster: topi
Processing monster: chilotse
Processing monster: neogunde
Processing monster: dwararct
Processing monster: kankwild
Processing monster: aeserpen
Processing monster: gianverb
Processing monster: shadecer
Processing monster: tortle
Processing monster: neotheli
Processing monster: kappa
Processing monster: chimera
Processing monster: dwaratha
Processing monster: afanc
Processing monster: gianwood
Processing monster: neshezu
Processing monster: chitine
Processing monster: agarat
Processing monster: gianzakh
Processing monster: translat
Processing monster: shadecla
Processing monster: dwarcrus
Processing monster: nethbird
Processing monster: karliki
Processing monster: choker
Processing monster: aggie
Processing monster: nicepona
Processing monster: gibbbroo
Processing monster: treaatha
Processing monster: dwarcude
Processing monster: chosenon
Processing monster: agrutha
Processing monster: kashehot
Processing monster: gibberli
Processing monster: treaevbr
Processing monster: nighagst
Processing monster: dwardaer
Processing monster: ahuizotl
Processing monster: chososio
Processing monster: gibbmogr
Processing monster: dwarderr
Processing monster: kech
Processing monster: shadelin
Processing monster: treaevil
Processing monster: airsenti
Processing monster: gibbmout
Processing monster: chraal
Processing monster: shadethe
Processing monster: keeper
Processing monster: treagrea
Processing monster: akikage
Processing monster: nighcraw
Processing monster: shadevar
Processing monster: giff
Processing monster: chrolily
Processing monster: dwarduer
Processing monster: treant
Processing monster: alaghi
Processing monster: kelpie
Processing monster: nighride
Processing monster: shadfien
Processing monster: chrovoid
Processing monster: gith
Processing monster: treaunde
Processing monster: albari
Processing monster: kender
Processing monster: dwarf
Processing monster: nighthag
Processing monster: shadhoun
Processing monster: chulcrix
Processing monster: githpira
Processing monster: treedark
Processing monster: alchplan
Processing monster: shadmast
Processing monster: nightmar
Processing monster: chuu
Processing monster: githyank
Processing monster: treesing
Processing monster: kenku
Processing monster: aleax
Processing monster: dwargull
Processing monster: nigmarbe
Processing monster: shadow
Processing monster: cildabri
Processing monster: trelon
Processing monster: kercpa
Processing monster: dwarhill
Processing monster: algoid
Processing monster: cilops
Processing monster: shadpers
Processing monster: githzera
Processing monster: tren
Processing monster: dwarmoun
Processing monster: alguduir
Processing monster: cinnavix
Processing monster: tressym
Processing monster: nigmarco
Processing monster: dwarthei
Processing monster: kestreke
Processing monster: cistfien
Processing monster: alhoon
Processing monster: shadslow
Processing monster: giwoitis
Processing monster: tribgath
Processing monster: dwarurdu
Processing monster: nigmarma
Processing monster: khaasta
Processing monster: trilloch
Processing monster: clamgian
Processing monster: shadspun
Processing monster: aljahar
Processing monster: dwarutuc
Processing monster: gkloklok
Processing monster: khargra
Processing monster: trin
Processing monster: clamgiav
Processing monster: allura
Processing monster: nigshams
Processing monster: shadstaf
Processing monster: dwarwild
Processing monster: gladliza
Processing monster: triton
Processing monster: cloaker
Processing monster: almiraj
Processing monster: kholiath
Processing monster: glittman
Processing monster: dwarzakh
Processing monster: troglody
Processing monster: killmoul
Processing monster: cloaksea
Processing monster: ambuchar
Processing monster: dweoborn
Processing monster: nigshato
Processing monster: glitworm
Processing monster: shadstee
Processing monster: cloalord
Processing monster: trolfire
Processing monster: killspru
Processing monster: amiqraso
Processing monster: nigterma
Processing monster: eagle
Processing monster: glomus
Processing monster: shadwarr
Processing monster: kindori
Processing monster: cloaresp
Processing monster: amiskwia
Processing monster: trolgray
Processing monster: gloowing
Processing monster: niiath
Processing monster: eaofknow
Processing monster: ammut
Processing monster: cloashad
Processing monster: kirin
Processing monster: shadwrat
Processing monster: glouras
Processing monster: trolhoun
Processing monster: nikaal
Processing monster: amphiter
Processing monster: cloaunde
Processing monster: echo
Processing monster: gluttsea
Processing monster: kirre
Processing monster: shalarin
Processing monster: troll
Processing monster: niktoo
Processing monster: amphpois
Processing monster: cloudray
Processing monster: gnasher
Processing monster: shammoun
Processing monster: echyan
Processing monster: kitskasu
Processing monster: ningyo
Processing monster: anadjiin
Processing monster: kizoku
Processing monster: gnaswing
Processing monster: clubthor
Processing monster: eeaar
Processing monster: shansao
Processing monster: nishruu
Processing monster: ancestor
Processing monster: cluricha
Processing monster: gnoll
Processing monster: klaatah
Processing monster: shapeshi
Processing monster: ancimari
Processing monster: nixie
Processing monster: clwohorr
Processing monster: gnome
Processing monster: eel
Processing monster: trollega
Processing monster: klar
Processing monster: andeloid
Processing monster: nocnitsa
Processing monster: trollice
Processing monster: eelgimor
Processing monster: kluzd
Processing monster: sharg
Processing monster: clwoswma
Processing monster: noctral
Processing monster: anemongs
Processing monster: trolmuta
Processing monster: klyndes
Processing monster: eelsuper
Processing monster: shargugh
Processing monster: gnomspri
Processing monster: egarus
Processing monster: noran
Processing monster: trolphaz
Processing monster: kna
Processing monster: cockatri
Processing monster: shark
Processing monster: gnomtink
Processing monster: anglfish
Processing monster: nosferat
Processing monster: trolsnow
Processing monster: knighter
Processing monster: coffcorp
Processing monster: einheria
Processing monster: sharkath
Processing monster: angreden
Processing monster: trolston
Processing monster: kobodrmo
Processing monster: coldwoma
Processing monster: anguilii
Processing monster: sharkkin
Processing monster: goatman
Processing monster: ekimmu
Processing monster: nuckalav
Processing monster: colossus
Processing monster: trosip
Processing monster: eladgfir
ProcessingThe command '/bin/bash -c ./LomionReader' returned a non-zero code: 139
```

**Comment by decheine** [comment]
Thank you for raising this issue, it's strange that you're experiencing this. A few questions:
1. Are you running this from a fresh copy of the repo? Like no modifications to monster pages or existing monsters. Because if something goes wrong during parsing an error like this can happen. 
2. Can you try running it again without sudo? It shouldn't need su permissions and that could cause problems when reading files.

Try it again without sudo and let me know if it's any different. It ran on my Windows machine and WSL Ubuntu, so I'm not sure what else could be causing the issue.

**Comment by decheine** [comment]
If it keeps crashing, in the file `/harvester/src/main.cpp` line 63 change `harvester.HarvestAll(true);` to `harvester.HarvestAll(false);` and rebuild. This will make the harvester run in sequence (1 thread) instead of multithreaded. This will also allow you to see which monster page it trips up on. 

I ran it in WSL with sudo and it still ran fine. I was hoping the behavior would always be the same with Docker but I guess not.

**Comment by larsjsol** [comment]
After digging around for a bit with a debugger it seems like it get stuck while running some regular expressions on `eldetevi`.

I got it working by changing this: https://github.com/larsjsol/complete-compendium/commit/85e4a891a3e9c33c8477a9d1f631fa5bc6a16905

As far as I can tell `splitTestResult` is not used at any later point in `Harvester::GetFullBody()`.

**Comment by larsjsol** [comment]
I'm just poking around, but let me know if you would like a pull request

**Comment by decheine** [comment]
Oh interesting, nice finding that! I wasn't able to reproduce it but did removing that statement and only that statement resolve the error? Yeah it looks like the value of the variable isn't used after that statement so it's fine to remove, it was likely a leftover from debugging old code. My apologies for the absolute dumpster fire of c++, I made it years ago and just slapped it together so I could get something working. 

I can make that replacement quick, I also pushed something earlier today that you'll need to fetch. 

Let me know if you get through the rest of the steps with or without issue.
